builder.go

```diff
@@ -0,0 +1,140 @@
+package grpcwot
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/Interactions-HSG/grpcwot/pkg/protofmt"
+	"github.com/emicklei/proto"
+	"github.com/linksmart/thing-directory/wot"
+)
+
+type builder struct {
+	td   wot.ThingDescription
+	ds   map[string]*wot.DataSchema
+	ip   string
+	port int
+}
+
+func newBuilder(ip string, port int) *builder {
+	return &builder{
+		td:   wot.ThingDescription{},
+		ds:   map[string]*wot.DataSchema{},
+		ip:   ip,
+		port: port,
+	}
+}
+
+// GetIRI returns the target IRI for the RPC
+func (b *builder) GetIRI(rpcName string) string {
+	return fmt.Sprintf("http://%s:%d/%s/%s", b.ip, b.port, b.td.Title, rpcName)
+}
+
+// HandleService assigns the Title for the resulting TD
+func (b *builder) HandleService(s *proto.Service) {
+	b.td.Title = s.Name
+}
+
+// HandleRPC constructs Interaction Affordances with prefetched DataSchema
+func (b *builder) HandleRPC(r *proto.RPC) {
+	// TODO: affordance classification should happen here
+	// putting all RPC to Actions for now
+	if b.td.Actions == nil {
+		b.td.Actions = map[string]wot.ActionAffordance{}
+	}
+	affordance := wot.ActionAffordance{}
+	affordance.Input = *b.ds[r.RequestType]
+	affordance.Output = *b.ds[r.ReturnsType]
+	affordance.Forms = []wot.Form{
+		{
+			Href:        b.GetIRI(r.Name),
+			ContentType: "application/grpc+proto",
+			Op:          []string{"writeproperty"},
+		},
+	}
+
+	b.td.Actions[r.Name] = affordance
+}
+
+// HandleMessage build a DataSchema: https://www.w3.org/TR/wot-thing-description/#dataschema
+// from a Message in the protobuf definition
+func (b *builder) HandleMessage(m *proto.Message) {
+	if _, ok := b.ds[m.Name]; !ok {
```

Comment:
Do nothing (there should be no duplicates).

Reply:
Ah, then there's a bug. The following for block needs to be left out if there're duplicate Messages.

Reply:
I think we have to distinguish:
- If the message is nested, it would have to be added. 
- If it is not nested the hole proto would be invalid and we should maybe break the process, otherwise it is not clear, what message is referred to by using the message's name in rpc
Can we keep that problem in mind and come back to it, when we deal with nested messages?